src/test.rs

```diff
@@ -543,6 +547,27 @@ impl<S: 'static> TestRequest<S> {
         self
     }
 
+    /// set cookie of this request
+    pub fn cookie(mut self, cookie: Cookie<'static>) -> Self {
+        let mut should_insert = true;
+        let mut cookies = match self.cookies {
```

Comment:
I think it would be better to match by mut reference and insert inside `Some(ref mut cookies)` instead of moving out and moving in again

Reply:
Thanks for you advice, all done.